src/ekat/ekat_pack_kokkos.hpp

```diff
@@ -409,6 +471,54 @@ void device_to_host(const Kokkos::Array<typename ViewT::value_type::scalar*, N>&
   }
 }
 
+// 3d - set do_transpose to true if host data is going to fortran
+template <typename SizeT, size_t N, typename ViewT>
+void device_to_host(const Kokkos::Array<typename ViewT::value_type::scalar*, N>& data,
+                    const Kokkos::Array<SizeT, N>& dim1_sizes,
+                    const Kokkos::Array<SizeT, N>& dim2_sizes,
+                    const Kokkos::Array<SizeT, N>& dim3_sizes,
+                    Kokkos::Array<ViewT, N>& views,
+                    bool do_transpose=false)
+{
+  using PackT = typename ViewT::value_type;
+  using ScalarT = typename PackT::scalar;
+
+  std::vector<ScalarT> tdata;
+  for (size_t n = 0; n < N; ++n) {
+    const size_t dim1_size = static_cast<size_t>(dim1_sizes[n]);
+    const size_t dim2_size = static_cast<size_t>(dim2_sizes[n]);
+    const size_t dim3_size = static_cast<size_t>(dim3_sizes[n]);
+    const size_t npack = views[n].extent(2);
+    const auto host_view = Kokkos::create_mirror_view(views[n]);
+    Kokkos::deep_copy(host_view, views[n]);
+
+    ScalarT* the_data = nullptr;
+    if (do_transpose) {
+      tdata.reserve(dim1_size * dim2_size * dim3_size);
+      the_data = tdata.data();
+    }
+    else {
+      the_data = data[n];
+    }
+
+    for (size_t i = 0; i < dim1_size; ++i) {
+      for (size_t k = 0; k < dim2_size; ++k) {
+        for (size_t p = 0; p < npack; ++p) {
+          const size_t num_scalars_this_col = p*PackT::n;
+          const size_t scalar_offset = i*(dim2_size*dim3_size) + k*dim3_size + num_scalars_this_col;
+          for (size_t s = 0; s < PackT::n && num_scalars_this_col+s < dim3_size; ++s) {
+            the_data[scalar_offset + s] = host_view(i, k, p)[s];
```

Comment:
Same commend (backwards) as before: why not just using memcpy, rather than the three for loops (which are arguably harder to follow)?